addons/web/static/tests/views/kanban_tests.js

```diff
@@ -3728,6 +3728,61 @@ QUnit.module('Views', {
         kanban.destroy();
     });
 
+    QUnit.test('nocontent helper after adding a record(kanban with progressbar)', async function (assert) {
+        assert.expect(3);
+
+        const kanban = await createView({
+            View: KanbanView,
+            model: 'partner',
+            data: this.data,
+            arch: `<kanban >
+                    <field name="product_id"/>
+                    <progressbar field="foo" colors='{"yop": "success", "gnap": "warning", "blip": "danger"}' sum_field="int_field"/>
+                    <templates>
+                      <t t-name="kanban-box">
+                        <div><field name="foo"/></div>
+                      </t>
+                    </templates>
+                </kanban>`,
+            groupBy: ['product_id'],
+            domain: [['foo', '=', 'abcd']],
+            mockRPC: function (route, args) {
+                if (args.method === 'web_read_group') {
+                    const result = {
+                        groups: [
+                            { __domain: [['product_id', '=', 3]], product_id_count: 0, product_id: [3, 'hello'] },
+                        ],
+                    };
+                    return Promise.resolve(result);
+                }
+                return this._super.apply(this, arguments);
+            },
+            viewOptions: {
+                action: {
+                    help: "No content helper",
+                },
+            },
+        });
+
+        assert.strictEqual(kanban.$('.o_view_nocontent').length, 1,
+            "there should be a nocontent helper");
+
+        // add a record
+        await testUtils.dom.click(kanban.$('.o_kanban_quick_add'));
+        await testUtils.fields.editInput(kanban.$('.o_kanban_quick_create .o_input'), 'twilight sparkle');
+        await testUtils.dom.click(kanban.$('button.o_kanban_add'));
+
+        assert.strictEqual(kanban.$('.o_view_nocontent').length, 0,
+            "there should be no nocontent helper (there is now one record)");
+
+        // cancel quick create
+        await testUtils.dom.click(kanban.$('button.o_kanban_cancel'));
+        assert.strictEqual(kanban.$('.o_view_nocontent').length, 0,
+            "there should be no nocontent helper after cancelling quick create");
```

Comment:
This isn't exactly the scenario you described in the commit message. Do you confirm we faced the same issue in this situation?

Reply:
Yes there was an issue, but a secondary one:
- displaying the quick create form forces the nocontent helper to be removed
- cancelling the quick create toggles the nocontent, but this time relies on datapoint.count